esphome/components/light/light_color_values.h

```diff
@@ -188,12 +188,23 @@ class LightColorValues {
     }
   }
 
-  /// Convert these light color values to an CWWW representation with the given parameters.
-  void as_ct(float color_temperature_cw, float color_temperature_ww, float *color_temperature) const {
-    if (this->color_mode_ & ColorCapability::COLOR_TEMPERATURE) {
+  /// Convert these light color values to a CT+BR representation with the given parameters.
+  void as_ct(float color_temperature_cw, float color_temperature_ww, float *color_temperature, float *white_brightness,
+             float gamma = 0) const {
+    if (this->color_mode_ & ColorCapability::RGB) {  // Handles lights in RGB_COLOR_TEMPERATURE mode.
+      const float ct_level =
+          (this->color_temperature_ - color_temperature_cw) / (color_temperature_ww - color_temperature_cw);
+      *color_temperature = ct_level;
+      const float brightness = gamma_correct(this->state_ * this->brightness_ * this->white_, gamma);
+      *white_brightness = brightness;
+    } else if (this->color_mode_ &
+               ColorCapability::COLOR_TEMPERATURE) {  // Handles lights in COLOUR_TEMPERATURE only only mode.
       const float ct_level =
           (this->color_temperature_ - color_temperature_cw) / (color_temperature_ww - color_temperature_cw);
       *color_temperature = ct_level;
+      this->as_brightness(white_brightness, gamma);
```

Comment:
Maybe something like this is a bit cleaner (less code duplication)?
```suggestion
    const float white_level = this->color_mode_ & ColorCapability::RGB ? this->white_ : 1;
    if (this->color_mode_ & ColorCapability::COLOR_TEMPERATURE) {
      const float ct_level =
          (this->color_temperature_ - color_temperature_cw) / (color_temperature_ww - color_temperature_cw);
      *color_temperature = ct_level;
      *white_brightness = gamma_correct(this->state_ * this->brightness_ * white_level, gamma);
```

Reply:
I knew there would be a simpler way to do this...haha
Going to sleep now :zzz: